scripts/protocversioncheck.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+PROTOC_CMD=${PROTOC_CMD:-protoc}
+PROTOC_VERSION_MIN="$1"
+echo "==> Checking that protoc is at version $1..."
+
+PROTOC_VERSION=$($PROTOC_CMD --version | grep -o '[0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?')
+
+IFS="." read -r -a PROTOC_VERSION_ARR <<< "$PROTOC_VERSION"
+IFS="." read -r -a PROTOC_VERSION_REQ <<< "$PROTOC_VERSION_MIN"
+
+if [[ ${PROTOC_VERSION_ARR[0]} -lt ${PROTOC_VERSION_REQ[0]} ||
+  ( ${PROTOC_VERSION_ARR[0]} -eq ${PROTOC_VERSION_REQ[0]} &&
+  ( ${PROTOC_VERSION_ARR[1]} -lt ${PROTOC_VERSION_REQ[1]} ||
+  ( ${PROTOC_VERSION_ARR[1]} -eq ${PROTOC_VERSION_REQ[1]} && ${PROTOC_VERSION_ARR[2]} -lt ${PROTOC_VERSION_REQ[2]} )))
+  ]]; then
+  echo "protoc should be at $PROTOC_VERSION_MIN; found $PROTOC_VERSION."
```

Comment:
nit: Should we something like "at or above" to denote that this is the minimum version required?

Reply:
I think the protoc version should be pinned to a certain version since the auto generated files contain the protobuf related versions:
```
// versions:
  3 //      protoc-gen-go v1.28.1
  4 //      protoc        v3.21.5
```
That would cause confusion the same way as now.

Reply:
Maybe we can also add to the comment, that in case you have higher version (preferably the latest) update the Makefile with that version and rerun "make proto"?

Reply:
Fixed in https://github.com/hashicorp/vault/pull/17122/commits/3b1a99a7a09c8f9deda65df8d5a91bea0a33c7b8